src/home/Routes.tsx

```diff
@@ -0,0 +1,85 @@
+
+  return ( 
+      <Switch>
+        {
+          appTiles.map(appTile=> appTile.onPage).filter((value, index, self) => {
+            return self.indexOf(value) === index;
+          }).map(appTilePage=>{
+            return <Route key={appTilePage} exact path={appTilePage} component={Home} />
+          })
+        }
+        <Route exact path="/personal_api_keys" component={PersonalApiKeysTable} />
+        <Route exact path="/personal_api_keys/new" component={NewPersonalApiKey} />
+        <Route exact path="/personal_api_keys/:uuid" component={EditPersonalApiKey} />
+        <Route exact path="/data_management/rasters/sources" component={RasterSourceTable} />
+        <Route exact path="/data_management/rasters/layers" component={RasterLayerTable} />
+        <Route exact path="/data_management/wms_layers" component={WmsLayerTable} />
+        <Route exact path="/data_management/scenarios" component={ScenarioTable} />
+        <Route exact path="/data_management/labels/label_types" component={LabeltypesTable} />
+        <Route exact path="/data_management/rasters/sources/new" component={NewRasterSource} />
+        <Route exact path="/data_management/rasters/layers/new" component={NewRasterLayer} />
+        <Route exact path="/data_management/rasters/sources/:uuid" component={EditRasterSource} />
+        <Route exact path="/data_management/rasters/layers/:uuid" component={EditRasterLayer} />
+        <Route exact path="/data_management/rasters/:id/data" component={UploadRasterData} />
+        <Route exact path="/data_management/wms_layers/new" component={NewWmsLayer} />
+        <Route exact path="/data_management/wms_layers/:id" component={EditWmsLayer} />
+        <Route exact path="/data_management/scenarios/:uuid" component={EditScenario} />
+        <Route exact path="/data_management/labels/label_types/:uuid" component={EditLabeltype} />
+        <Route exact path="/alarms/notifications" component={NotificationsApp} />
+        <Route exact path="/alarms/notifications/new" component={NewNotificationApp} />
+        <Route exact path="/alarms/notifications/:id" component={EditNotificationApp} />
+        <Route exact path="/alarms/groups" component={AlarmGroupsApp} />
+        <Route exact path="/alarms/groups/new" component={NewAlarmGroupApp} />
+        <Route exact path="/alarms/groups/:id" component={AlarmGroupsDetail} />
+        <Route exact path="/alarms/contacts" component={AlarmContactApp} />
+        <Route exact path="/alarms/contacts/new" component={NewContactApp} />
+        <Route exact path="/alarms/contacts/:id" component={ContactDetail} />
+        <Route exact path="/alarms/templates" component={AlarmTemplatesApp} />
+        <Route exact path="/alarms/templates/new" component={NewTemplateApp} />
+        <Route exact path="/alarms/templates/:id" component={AlarmTemplatesDetail} />
```

Comment:
I thought the map function already rendered all the Routes (but without the component). Is it a duplication of what is returned from the map function?

Reply:
The map function renders all routes that show the 'home' component.
The home components renders the tile based layout.

The hardcoded routes below are all for rendering no-tile based layouts (tables, forms etcetera)

Reply:
I feel the map function should render a routing component for that route. That can then have its own router that chooses which part to render, home, table, form et cetera. Use nested routing.

Reply:
I think nested routing could be good, but I have not done it for the following 2 reasons:
- It will be harder to generate the routing from a list of configured tiles (assuming that the the list of tiles is itself a flat list and thus not a json tree structure)
- It will require more knowledge from the routing library. I would need to investigate.

My plan was to keep the list of tiles a flat list with its own database table that a user can then configure on the management page. Therefore I prefer not to use a nested structure. Is this acceptable for you? Some changes to the url will require a bit more work, but nothing too complicated I guess.

Reply:
- The question is to either hardcode this list here, or inside components. It's hard coded both ways.

- A user would probably configure on the level of the appTiles list, and not on individual parts (no users wants to configure that he wants to have the alarms, but not that one form on the alarms subpages). Same general reason why you configured some of them in the appTiles list and not all of them. So it wouldn't be touched by more configuration by the user.

- Nesting with hooks is very simple ( https://reactrouter.com/web/example/nesting ) but I don't know how it was with classes exactly.